src/lib.rs

```diff
@@ -168,24 +337,150 @@ pub use restate_sdk_macros::service;
+/// **Note: Workflow retention time**:
+/// The retention time of a workflow execution is 24 hours after the finishing of the `run` handler.
+/// After this timeout any [K/V state][crate::context::ContextReadState] is cleared, the workflow's shared handlers cannot be called anymore, and the Durable Promises are discarded.
+/// The retention time can be configured via the [Admin API](https://docs.restate.dev//references/admin-api/#tag/service/operation/modify_service) per Workflow definition by setting `workflow_completion_retention`.
+///
+/// ## Implementing workflows
+/// Have a look at the code example to get a better understanding of how workflows are implemented:
 ///
 /// ```rust,no_run
 /// use restate_sdk::prelude::*;
 ///
 /// #[restate_sdk::workflow]
-/// trait Billing {
-///     async fn run() -> Result<u64, TerminalError>;
+/// pub trait SignupWorkflow {
+///     async fn run(req: String) -> Result<bool, HandlerError>;
 ///     #[shared]
-///     async fn get_status() -> Result<String, TerminalError>;
+///     async fn click(click_secret: String) -> Result<(), HandlerError>;
+///     #[shared]
+///     async fn get_status() -> Result<String, HandlerError>;
+/// }
+///
+/// pub struct SignupWorkflowImpl;
+///
+/// impl SignupWorkflow for SignupWorkflowImpl {
+///     async fn run(&self, mut ctx: WorkflowContext<'_>, email: String) -> Result<bool, HandlerError> {
+///
+///         let secret = ctx.rand_uuid().to_string();
+///         ctx.run(|| send_email_with_link(email.clone(), secret.clone())).await?;
+///         ctx.set("status", "Email sent".to_string());
+///
+///         let click_secret = ctx.promise::<String>("email.clicked").await?;
+///         ctx.set("status", "Email clicked".to_string());
+///
+///         Ok(click_secret == secret)
+///     }
+///     async fn click(&self, ctx: SharedWorkflowContext<'_>, click_secret: String) -> Result<(), HandlerError> {
+///         ctx.resolve_promise::<String>("email.clicked", click_secret);
+///         Ok(())
+///     }
+///     async fn get_status(&self, ctx: SharedWorkflowContext<'_>) -> Result<String, HandlerError> {
+///         Ok(ctx.get("status").await?.unwrap_or("unknown".to_string()))
+///     }
+/// }
+/// # async fn send_email_with_link(email: String, secret: String) -> Result<(), HandlerError> {
+/// #    Ok(())
+/// # }
+///
+/// #[tokio::main]
+/// async fn main() {
+///     tracing_subscriber::fmt::init();
+///     HttpServer::new(Endpoint::builder().bind(SignupWorkflowImpl.serve()).build())
+///         .listen_and_serve("0.0.0.0:9080".parse().unwrap())
+///         .await;
 /// }
 /// ```
+///
+/// ### The run handler
+///
+/// Every workflow needs a `run` handler.
+/// This handler has access to the same SDK features as Service and Virtual Object handlers.
+/// In the example above, we use [`ctx.run`][crate::context::ContextSideEffects] to log the sending of the email in Restate and avoid re-execution on replay.
```

Comment:
can you link to the actual function rather than only the trait? Same comment for many of the other cases.